fix(web-app): gracefully handle prismjs crashes

jharvey10 opened the pull request:
Fixes the issue with the schemas page crashing in a minified build.

#### Changelog

**New**

- include a prism download from prismjs.com (recommended approach until full esm support is added)
- - highlight code via prism in a useEffect block in the code component

**Changed**

- ‼️ switch to non-wrapping code blocks
  - [x] verify that this behavior is okay and is an improvement

**Removed**

- remove port number from next dev command so it can be manually specified

#### Testing / reviewing

Check visual style of code blocks across a variety of pages to ensure it looks the same as before (except for the new lack of wrapping)

jharvey10 committed:
fix(web-app): gracefully handle prismjs crashes

- highlight code via prism in a useEffect block in the code component
- switch to non-wrapping code blocks
- include a prism download from prismjs.com (recommended approach until full esm support is added)
- remove port number from next dev command so it can be manually specified

mattrosno commented on services/web-app/styles/_prism-theme.scss:
```diff
@@ -0,0 +1,160 @@
+  opacity: 0.7;
+}
+
+/* LANGUAGE-SPECIFIC OVERRIDES */
+// .language-css .token.important {
+//   color: unset;
+// }
```
Did you mean to remove this?

jdharvey-ibm replied:
I wanted to keep it there for future reference in case we ever need to add custom language colors. I can remove it if you'd prefer that.

mattrosno replied:
Oh, no preference, maybe a comment would help "for future reference, if we need to... here's an example of what we'd need to do...".

jdharvey-ibm commented on services/web-app/public/prism.js:
```diff
@@ -0,0 +1,8 @@
+/* PrismJS 1.29.0
```
I definitely thought about that and we can certainly add it. I took what their site provided verbatim, assuming that if they wanted their license in the minified version, they would have included it. The URL at the top does link back to their site, which has a paper trail to their license and the mechanism by which we downloaded it.

mattrosno replied:
Or, we save both prism.js and prism.min.js so we can save the MIT comment, and then only use the minified version in the web app.

jdharvey-ibm replied:
👍 I can add the non-minified version.

jharvey10 committed:
fix(web-app): move prismjs files to vendor folder

- also include non-minified prism since it contains their license
- update ignore files accordingly

jharvey10 committed:
docs(web-app): improve sass docs for prism color overrides

alisonjoseph commented on services/web-app/components/code/code.module.scss:
```diff
@@ -27,22 +26,11 @@
-  color: theme.$text-primary;
 }
 
-// override carbon code styles
-.code :global(.cds--snippet-container) {
-  width: 100%;
-  // override carbon inline style
-  min-height: 24px !important;
-
-  pre {
-    padding-bottom: 0;
-  }
-
-  code {
-    background-color: theme.$layer-01;
-  }
```
I think we need these styles?

alisonjoseph reviewed (changes requested):
I'm seeing a lot styling errors. I think it has to do with some of the lines removed 33-45 in code.module.scss file? Also looks like we've lost syntax highlighting on some of the code blocks that had it previously. (was testing this page http://localhost:3000/libraries/carbon-react/latest/pages/tutorial/step-2)

<img width="1240" alt="Screen Shot 2022-09-20 at 7 29 46 AM" src="https://user-images.githubusercontent.com/2753488/191258069-1b6b6a9a-e6df-450d-adda-5cd3dd83449d.png">
<img width="1310" alt="Screen Shot 2022-09-20 at 7 30 08 AM" src="https://user-images.githubusercontent.com/2753488/191258078-45f5d809-00b7-4d46-b6b1-ec3a988c6f7b.png">
<img width="775" alt="Screen Shot 2022-09-20 at 7 30 30 AM" src="https://user-images.githubusercontent.com/2753488/191258086-b10f64fb-e590-4cf7-99e5-5bdb99c248f2.png">
<img width="761" alt="Screen Shot 2022-09-20 at 7 30 51 AM" src="https://user-images.githubusercontent.com/2753488/191258091-198df95b-4c3d-43c8-93b7-cfd538b19164.png">

alisonjoseph commented:
> ‼️ switch to non-wrapping code blocks 

Verifying this is what we want and a bug on our live site 👍

jharvey10 committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into fix-schemas-page-crash

jharvey10 committed:
fix(web-app): fix code snippet styling

- add more prism languages
- remove js prism theme file

alisonjoseph commented on services/web-app/components/code/code.module.scss:
```diff
@@ -9,54 +9,41 @@
 @use '@carbon/react/scss/motion' as motion;
-/* stylelint-disable declaration-no-important */
-
-// scoped to div code to override global reset styles
-:global(div) code {
-  padding: 0 spacing.$spacing-03;
-  background-color: theme.$layer-accent;
-  border-radius: 4px;
-  font-size: 0.875em;
-}
```
Missed this before, these styles are styling the base inline code in markdown, we aren't translating them to the inline Carbon code snippet for some reason. Probably should be vs. creating our own styles here, trying to remember why I did it this way. 

<img width="1152" alt="Screen Shot 2022-09-20 at 2 06 02 PM" src="https://user-images.githubusercontent.com/2753488/191343266-ddc55a19-7d92-4e13-b173-6040d65fdd2e.png">

jharvey10 committed:
fix(web-app): use inline code snippet in mdx pages

jharvey10 committed:
fix(mdx-components): export component props interfaces

jharvey10 committed:
fix(web-app): add markdown to prism langs

francinelucca reviewed (changes requested):
Looks Great!
Only thing is I see a dark background on md breakpoint that I don't see on prod 
![image](https://user-images.githubusercontent.com/40550942/191614202-8ff92747-d487-4729-a32e-55cd502e0450.png)

alisonjoseph commented on services/web-app/components/mdx/components.js:
```diff
@@ -63,6 +63,9 @@ const components = {
   table: PageTable,
   pre: Code,
+  code: ({ children }) => (
+    <mdxComponents.CodeSnippet type="inline">{children}</mdxComponents.CodeSnippet>
```
Nit-picky, but should we pull this out into its own component?

jdharvey-ibm replied:
I definitely think we should! We could add a `Code` and `InlineCode` component to the mdx-components package. As we keep adding stuff over there, this file will get smaller to reflect that. Would that work? Or would you prefer to address it in this PR first and then move it to the mdx-components package as a second step?

alisonjoseph replied:
I'm fine with doing it when we move the Code component over to the mdx-components package.

alisonjoseph reviewed (approved):
Working great 🥳  just one small question/comment

jharvey10 committed:
fix(web-app): fix regression of tablet code snippet max width

francinelucca reviewed (approved):
✨

jharvey10 committed:
docs(web-app): add license header to prism theme
